pkg/instancetype/instancetype.go

```diff
@@ -1221,13 +1221,17 @@ func applyDiskPreferences(preferenceSpec *instancetypev1beta1.VirtualMachinePref
 	}
 }
 
+func isInterfaceBindingUnset(vmiIface *virtv1.Interface) bool {
+	return vmiIface.Masquerade == nil && vmiIface.Bridge == nil && vmiIface.Slirp == nil && vmiIface.SRIOV == nil && vmiIface.Macvtap == nil && vmiIface.Passt == nil
+}
+
```

Comment:
Thanks, nice

Can we use something generic ? 
maybe something like this (didn't try it, just to show the idea)

```
func areAllFieldsNil(ifaceBindingMethod *InterfaceBindingMethod) bool {
	val := reflect.ValueOf(ifaceBindingMethod).Elem()
	return val.IsZero()
}
```

Reply:
Yup sorry missed `InterfaceBindingMethod` was being used, updated.

Reply:
would panic if nil `panic: reflect: call of reflect.Value.IsZero on zero Value`
do we know it is not nil ?
otherwise we need please
```
	if ifaceBindingMethod == nil {
		return true
	}
```

Reply:
I don't think it's ever `nil` when `inlined`. The existing test interfaces should cover this as neither have a binding mode set at the moment and we don't panic:

https://github.com/kubevirt/kubevirt/blob/178a70fc9203f6fa19e0bd6b7970f64429c42238/pkg/instancetype/instancetype_test.go#L1445-L1450

I'll start refactoring these tests shortly under the following issue FWIW, they really need some love:

https://github.com/kubevirt/kubevirt/issues/10175

Reply:
Ok thanks, and what about the `&` can it be removed ? or it is a must

Reply:
It's required as you can't call `Elem` on a struct value, only Interface or Pointer :

`reflect: call of reflect.Value.Elem on struct Value`